packages/oauth/oauth-provider/src/token/token-manager.ts

```diff
@@ -33,59 +32,72 @@ import { RequestMetadata } from '../lib/http/request.js'
 import { dateToEpoch, dateToRelativeSeconds } from '../lib/util/date.js'
 import { callAsync } from '../lib/util/function.js'
 import { OAuthHooks } from '../oauth-hooks.js'
+import { Sub } from '../oidc/sub.js'
 import { Code, isCode } from '../request/code.js'
+import { SignedTokenPayload } from '../signer/signed-token-payload.js'
 import { Signer } from '../signer/signer.js'
 import {
   generateRefreshToken,
   isRefreshToken,
   refreshTokenSchema,
 } from './refresh-token.js'
-import { TokenClaims } from './token-claims.js'
 import { TokenData } from './token-data.js'
-import {
-  TokenId,
-  generateTokenId,
-  isTokenId,
-  tokenIdSchema,
-} from './token-id.js'
+import { TokenId, generateTokenId, isTokenId } from './token-id.js'
 import { TokenInfo, TokenStore } from './token-store.js'
 import {
   VerifyTokenClaimsOptions,
   VerifyTokenClaimsResult,
   verifyTokenClaims,
 } from './verify-token-claims.js'
 
-export type AuthenticateTokenIdResult = VerifyTokenClaimsResult & {
-  tokenInfo: TokenInfo
-}
+export { AccessTokenMode, Signer }
+export type { OAuthHooks, TokenStore, VerifyTokenClaimsResult }
 
 export class TokenManager {
   constructor(
     protected readonly store: TokenStore,
     protected readonly signer: Signer,
     protected readonly hooks: OAuthHooks,
-    protected readonly accessTokenType: AccessTokenType,
+    protected readonly accessTokenMode: AccessTokenMode,
     protected readonly tokenMaxAge = TOKEN_MAX_AGE,
   ) {}
 
   protected createTokenExpiry(now = new Date()) {
     return new Date(now.getTime() + this.tokenMaxAge)
   }
 
-  protected useJwtAccessToken(account: Account) {
-    if (this.accessTokenType === AccessTokenType.auto) {
-      return this.signer.issuer !== account.aud
-    }
+  protected async buildAccessToken(
+    tokenId: TokenId,
+    account: Account,
+    client: Client,
+    parameters: OAuthAuthorizationRequestParameters,
+    options: {
+      now: Date
+      expiresAt: Date
+    },
+  ): Promise<OAuthAccessToken> {
+    return this.signer.createAccessToken({
+      sid: tokenId,
+      sub: account.sub,
+      exp: dateToEpoch(options.expiresAt),
+      iat: dateToEpoch(options.now),
+      cnf: parameters.dpop_jkt ? { jkt: parameters.dpop_jkt } : undefined,
 
-    return this.accessTokenType === AccessTokenType.jwt
+      ...(this.accessTokenMode !== AccessTokenMode.stateless && {
+        aud: account.aud,
+        scope: parameters.scope,
+        // https://datatracker.ietf.org/doc/html/rfc8693#section-4.3
+        client_id: client.id,
+      }),
```

Comment:
I would expect these to be part of the stateless token but not the light token.

Reply:
right !